plugins/mimecast/komand_mimecast/tasks/monitor_siem_logs/task.py

```diff
@@ -118,15 +127,51 @@ def _filter_and_sort_recent_events(self, task_output: List[Dict[str, Any]], cut_
             self.logger.info(f"Latest event time returned from Mimecast logs: {latest_time}")
         return task_output
 
-    def _get_filter_time(self, custom_config: Dict[str, int]) -> int:
+    def _get_filter_time(self, custom_config: Dict[str, int], normal_running_cutoff: bool = False) -> int:
         """
         Apply custom_config params (if provided) to the task. If a lookback value exists, it should take
-        precedence (this can allow a larger filter time), otherwise use the default value.
+        precedence (this can allow a larger filter time), In the initial running we want to use a cutoff of
+        24 hrs, but for normal running we want to use a look back of 7 days in the event that the task is
+        stuck or in an error state for more than 24hrs. Otherwise use the default value.
         :param custom_config: dictionary passed containing `default` or `lookback` values
+        :param normal_running_cutoff: if this is the initial run of the Mimecast tasks
         :return: filter time to be applied in `_filter_and_sort_recent_events`
         """
 
-        default, lookback = custom_config.get("default", CUTOFF), custom_config.get("lookback")
-        filter_value = lookback if lookback else default
-        self.logger.info(f"Task execution will be applying filter period of {filter_value} hours...")
-        return int(filter_value)
+        # if there is a custom config we first check for a custom date, then custom hours back and will use that first
+        if custom_config:
+
+            if custom_config.get("lookback", {}):
```

Comment:
minor but for readability could save lookback into a variable and the condition and filter_time creation could read nicer? same for cutoff_date ? saves accessing the key, values each time

```
lookback_date, cutoff = custom_config.get("lookback", {}), custom_config.get("cutoff", {})
cutoff_date, cutoff_hours = cutoff.get("date"), cutoff.get("hours")

if lookback:
   filter_time = datetime(lookback.get("year"), ...)
elif cutoff_hours:
  ....
elif cutoff_date:
 ....
```

Reply:
This has been updated to use variables